scripts/discourse-sync/client.py

```diff
@@ -0,0 +1,22 @@
+import pydiscourse
+
+# Client extends pydiscourse.DiscourseClient with new methods
+class Client(pydiscourse.DiscourseClient):
+    def lock_post(self, post_id, locked, **kwargs):
+        """
+        Lock a post from being edited
+        https://docs.discourse.org/#tag/Posts/operation/lockPost
+
+        Args:
+            post_id: integer
+            locked: bool
+        """
+        if bool(locked):
+            kwargs["locked"] = "true"
+        else:
+            kwargs["locked"] = "false"
```

Comment:
In Python you wouldn't normally explicitly case to `bool`, especially if you're just using it in an `if`. e.g.:

```suggestion
        if locked:
            kwargs["locked"] = "true"
        else:
            kwargs["locked"] = "false"
```

A lot of the time you'd find people writing it in one line like below, but this is definitely a style thing, either is fine:

```suggestion
        kwargs["locked"] = "true" if locked else "false"
```